src/features/transaction-wizard/data/common.ts

```diff
@@ -0,0 +1,107 @@
+import { ZERO_ADDRESS } from '@/features/common/constants'
+import { BuildableTransactionFormField, BuildableTransactionFormFieldType } from '../models'
+import { zfd } from 'zod-form-data'
+import { z } from 'zod'
+import { isAddress } from '@/utils/is-address'
+import { bigIntSchema, numberSchema } from '@/features/forms/data/common'
+
+const invalidAddressMessage = 'Invalid address'
+export const optionalAddressFieldSchema = zfd.text(z.string().optional()).refine((value) => (value ? isAddress(value) : true), {
+  message: invalidAddressMessage,
+})
+export const addressFieldSchema = zfd.text().refine((value) => (value ? isAddress(value) : true), {
+  message: invalidAddressMessage,
+})
+
+export const senderFieldSchema = { sender: addressFieldSchema }
+export const senderField = {
+  sender: {
+    label: 'Sender address',
+    description: 'Account to pay from. Sends the transaction and pays the fee',
+    type: BuildableTransactionFormFieldType.Account,
+    placeholder: ZERO_ADDRESS,
+  } satisfies BuildableTransactionFormField,
+}
+
+export const receiverFieldSchema = { receiver: addressFieldSchema }
+export const receiverField = {
+  receiver: {
+    label: 'Receiver address',
+    description: 'Account to pay to',
+    type: BuildableTransactionFormFieldType.Account,
+    placeholder: ZERO_ADDRESS,
+  } satisfies BuildableTransactionFormField,
+}
+
+export const noteFieldSchema = { note: zfd.text(z.string().optional()) }
+export const noteField = {
+  note: {
+    label: 'Note',
+    description: 'A note for the transaction',
+    type: BuildableTransactionFormFieldType.Text,
+  } satisfies BuildableTransactionFormField,
+}
+
+export const feeFieldSchema = {
+  fee: z
+    .object({
+      setAutomatically: z.boolean(),
+      value: numberSchema(z.number().min(0.001).optional()),
+    })
+    .superRefine((fee, ctx) => {
+      if (!fee.setAutomatically && !fee.value) {
+        ctx.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: 'Required',
+          path: ['value'],
+        })
+      }
+    }),
+}
```

Comment:
I'm wondering if we should refactor this to
```
export const feeFieldSchema = z
  .object({
    setAutomatically: z.boolean(),
    value: numberSchema(z.number().min(0.001).optional()),
  })
  .superRefine((fee, ctx) => {
    if (!fee.setAutomatically && !fee.value) {
      ctx.addIssue({
        code: z.ZodIssueCode.custom,
        message: 'Required',
        path: ['value'],
      })
    }
  })
```
so that the consumer can decide the field name.
For example, 
```
const paymentSchema = zfd.formData({
  ...
  fee: feeFieldSchema,
  ....
})
```
vs 
```
const paymentSchema = zfd.formData({
  ...feeFieldSchema,
})
```

Reply:
We can adjust as we see more scenarios. My thought here is that fee will always be the same across all transactions, so It'd be strange to be using a different key per transaction.